tests/test_config_command.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+use regex::Regex;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+#[test]
+fn test_config_list_single() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.add_config(
+        r###"
+    [user]
+    name = "Some User"
+    "###
+        .as_bytes(),
+    );
+
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user.name"]);
+    insta::assert_snapshot!(stdout, @r###"
+    user.name="Some User"
+    "###);
+}
+
+#[test]
+fn test_config_list_table() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.remove_env_var("JJ_EMAIL");
+    test_env.add_config(
+        r###"
+    [user]
+    name = "Some User"
+    email = "some.user@example.com"
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user"]);
+    // TODO: Remove sorting when config crate maintains deterministic ordering.
+    insta::assert_snapshot!(
+        stdout.lines().sorted().join("\n"),
+        @r###"
+    user.email="some.user@example.com"
+    user.name="Some User"
+    user.timestamp="2001-02-03T04:05:07+07:00"
+    "###);
+}
+
+#[test]
+fn test_config_list_array() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config(
+        r###"
+    test-array = [1, "b", 3.4]
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "test-array"]);
+    insta::assert_snapshot!(stdout, @r###"
+    test-array=[1, "b", 3.4]
+    "###);
+}
+
+#[test]
+fn test_config_list_inline_table() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config(
+        r###"
+        [[test-table]]
+        x = 1
+        [[test-table]]
+        y = ["z"]
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "test-table"]);
+    insta::assert_snapshot!(stdout, @r###"
+    test-table=[{x=1}, {y=["z"]}]
+    "###);
+}
+
+#[test]
+fn test_config_list_all() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.remove_env_var("JJ_EMAIL");
+    test_env.remove_env_var("JJ_OP_USERNAME");
+    test_env.remove_env_var("JJ_OP_HOSTNAME");
```

Comment:
Nit: Since some of these are overridden below (all but `JJ_EMAIL`, it looks like), there's probably no need to first remove them here

Reply:
The env vars take precedence, so if I don't remove them the add_config below has no effect.

Reply:
Actually, it's no longer needed anyway since I made the test stop depending on those config values. Reverted those related changes in TestEnvironment for now.